### PR TITLE
Freshpaint destination interface in Typescript

### PR DESCRIPTION
**Pull request by logason**

### CODE REVIEW

**Review by hugihlynsson — APPROVED** [review]
Nice!

**Comment by hugihlynsson** [comment]
Should we strip the types and add this to the JS examples too?

**Comment by hugihlynsson** [comment]
@logason I made some modifications. Care to review?

File: javascript/web/freshpaint.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    freshpaint.init();
+    freshpaint.page();
+  },
+
+  identify: (userId) => freshpaint.identify(userId),
+
+  logEvent: (eventName, eventProperties) =>
+    freshpaint.track(eventName, eventProperties),
+
+  setUserProperties: function (userId, userProperties) {
```
Comment (logason) [inline javascript/web/freshpaint.js:15]:
```suggestion
  setUserProperties: (userId, userProperties) => {
```

File: typescript/web/freshpaint.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Install the freshpaint SDK: https://documentation.freshpaint.io/readme/guides/quickstart/installing-freshpaint
+
+// Import the Avo Codegen file from the location you've chosen
+import { CustomDestination } from "../path/to/Avo";
```
Comment (logason) [inline typescript/web/freshpaint.ts:4]:
```suggestion
import { FreshpaintDestinationInterface } from "../path/to/Avo";
```

File: typescript/web/freshpaint.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+// Get the Freshpaint types from here: https://documentation.freshpaint.io/readme/guides/quickstart/how-do-i-install-freshpaint-with-typescript
+
+const freshpaintDestinationInterface: CustomDestination = {
```
Comment (logason) [inline typescript/web/freshpaint.ts:8]:
```suggestion
const freshpaintDestinationInterface: FreshpaintDestinationInterface = {
```

**Review by logason — COMMENTED** [review]
Two minor suggestions, otherwise LGTM (would hit the _Approve_ if I could (I'm the PR creator))

File: typescript/web/freshpaint.ts (hunk 2 above)
Comment (hugihlynsson) [reply typescript/web/freshpaint.ts:4]: I believe that the Avo Codegen just spits out a single interface for all the destination interfaces called `CustomDestination`. Does that make sense?

File: javascript/web/freshpaint.js (hunk 1 above)
Comment (hugihlynsson) [reply javascript/web/freshpaint.js]: Fixed!

File: typescript/web/freshpaint.ts (hunk 2 above)
Comment (hugihlynsson) [reply typescript/web/freshpaint.ts:4]: It would be cool to rename it to `Destination` at one point!

**Pull request merged**